graylog2-server/src/main/java/org/graylog/security/certutil/CertRenewalService.java

```diff
@@ -17,13 +17,15 @@
 package org.graylog.security.certutil;
 
 import org.apache.commons.lang3.tuple.Pair;
+import org.apache.commons.lang3.tuple.Triple;
 import org.graylog2.cluster.Node;
+import org.graylog2.cluster.preflight.DataNodeProvisioningConfig;
 
 import java.security.cert.X509Certificate;
 import java.util.List;
 
 public interface CertRenewalService {
     void checkCertificatesForRenewal();
     void initiateRenewalForNode(String nodeId);
-    List<Pair<Node, X509Certificate>> findNodesAndCertificates();
+    List<Triple<Node, DataNodeProvisioningConfig, X509Certificate>> findNodes();
```

Comment:
When I see a Triple I am thinking if this is the right moment to introduce a class/record with better interfaces for this data structure. Should I try to adapt the code slightly?

Reply:
sure, go ahead

Reply:
maybe pull down the `DataNode` record from `CertificateRenewalResource`?

Reply:
changed it